# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1264059710 18000
#      Thu Jan 21 02:41:50 2010 -0500
# Node ID a0665fa8798cec267257782aa91f38a8994dcd1f
# Parent  d354be89b2c37de8e8ddc6b00d79e83ae9a200a3
use symlink module from gnulib

diff --git a/ChangeLog b/ChangeLog
--- a/ChangeLog
+++ b/ChangeLog
@@ -1,8 +1,12 @@
+2010-01-21  John W. Eaton  <jwe@octave.org>
+
+	* bootstrap.conf (gnulib_modules): Include symlink in the list.
+
 2010-01-21  John W. Eaton  <jwe@octave.org>
 
 	* bootstrap.conf (gnulib_modules): Include link in the list.
 
 2010-01-21  John W. Eaton  <jwe@octave.org>
 
 	* bootstrap.conf (gnulib_modules): Include unlink in the list.
 
diff --git a/bootstrap.conf b/bootstrap.conf
--- a/bootstrap.conf
+++ b/bootstrap.conf
@@ -28,16 +28,17 @@ gnulib_modules="
   lstat
   mkdir
   mkfifo
   rename
   rmdir
   stat
   strftime
   strptime
+  symlink
   times
   unlink
   round
 "
 
 # Additional xgettext options to use.  Use "\\\newline" to break lines.
 XGETTEXT_OPTIONS=$XGETTEXT_OPTIONS'\\\
  --from-code=UTF-8\\\
diff --git a/liboctave/ChangeLog b/liboctave/ChangeLog
--- a/liboctave/ChangeLog
+++ b/liboctave/ChangeLog
@@ -1,8 +1,14 @@
+2010-01-21  John W. Eaton  <jwe@octave.org>
+
+	* lo-cutils.c (octave_symlink): New funtion.
+	* lo-utils.h: Provide decl.
+	* file-ops.cc (file_ops::symlink): Call octave_symlink.
+
 2010-01-21  John W. Eaton  <jwe@octave.org>
 
 	* lo-cutils.c (octave_link): New funtion.
 	* lo-utils.h: Provide decl.
 	* file-ops.cc (file_ops::link): Call octave_link.
 
 2010-01-21  John W. Eaton  <jwe@octave.org>
 
diff --git a/liboctave/file-ops.cc b/liboctave/file-ops.cc
--- a/liboctave/file-ops.cc
+++ b/liboctave/file-ops.cc
@@ -190,28 +190,23 @@ file_ops::symlink (const std::string& ol
 int
 file_ops::symlink (const std::string& old_name,
 		   const std::string& new_name, std::string& msg)
 {
   msg = std::string ();
 
   int status = -1;
 
-#if defined (HAVE_SYMLINK)
-
-  status = ::symlink (old_name.c_str (), new_name.c_str ());
+  status = octave_symlink (old_name.c_str (), new_name.c_str ());
 
   if (status < 0)
     {
       using namespace std;
       msg = ::strerror (errno);
     }
-#else
-  msg = NOT_SUPPORTED ("symlink");
-#endif
 
   return status;
 }
 
 // We provide a replacement for rename().
 
 int
 file_ops::readlink (const std::string& path, std::string& result)
diff --git a/liboctave/lo-cutils.c b/liboctave/lo-cutils.c
--- a/liboctave/lo-cutils.c
+++ b/liboctave/lo-cutils.c
@@ -54,16 +54,22 @@ Software Foundation, Inc.
 
 OCTAVE_API int
 octave_link (const char *old_name, const char *new_name)
 {
   return link (old_name, new_name);
 }
 
 OCTAVE_API int
+octave_symlink (const char *old_name, const char *new_name)
+{
+  return symlink (old_name, new_name);
+}
+
+OCTAVE_API int
 octave_mkdir (const char *name, mode_t mode)
 {
   return mkdir (name, mode);
 }
 
 OCTAVE_API int
 octave_mkfifo (const char *name, mode_t mode)
 {
diff --git a/liboctave/lo-utils.h b/liboctave/lo-utils.h
--- a/liboctave/lo-utils.h
+++ b/liboctave/lo-utils.h
@@ -49,16 +49,19 @@ extern OCTAVE_API std::string octave_fge
 extern OCTAVE_API std::string octave_fgets (std::FILE *, bool& eof);
 extern OCTAVE_API std::string octave_fgetl (std::FILE *, bool& eof);
 
 extern "C" OCTAVE_API int octave_gethostname (char *, int);
 
 extern "C" OCTAVE_API int
 octave_link (const char *old_name, const char *new_name);
 
+extern "C" OCTAVE_API int
+octave_symlink (const char *old_name, const char *new_name);
+
 extern "C" OCTAVE_API int octave_mkdir (const char *name, mode_t mode);
 
 extern "C" OCTAVE_API int octave_mkfifo (const char *name, mode_t mode);
 
 extern "C" OCTAVE_API void octave_qsort (void *base, size_t n, size_t size,
 			      int (*cmp) (const void *, const void *));
 
 extern "C" OCTAVE_API int octave_rmdir (const char *name);
